libraries/state_history/include/eosio/state_history/log.hpp

```diff
@@ -89,7 +89,7 @@ namespace state_history {
       fc::path retained_dir       = "retained";
       fc::path archive_dir        = "archive";
       uint32_t stride             = 1000000;
-      uint32_t max_retained_files = 10;
+      uint32_t max_retained_files = UINT32_MAX;
```

Comment:
The PR title and description say "change state-history-stride default", but here max_retained_files default is changed. Please correct the PR title and description.

You can add some explanation of this default to the command line help `max-retained-history-files`.